### PR TITLE
Assettree component

### PR DESCRIPTION
**Pull request by Kamillowl**

### CODE REVIEW

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-78.gearboxjs.preview.cogniteapp.com

**Comment by f1cognite** [comment]
In the "All data on mount" example I see an arrow for the node "VAL: Valhall plattform" but it seems it don't have any children. I can't expand the treeview.
Screenshot 2019-04-09 at 12 36 17

**Comment by Kamillowl** [comment]
> In the "All data on mount" example I see an arrow for the node "VAL: Valhall plattform" but it seems it don't have any children. I can't expand the treeview.
> Screenshot 2019-04-09 at 12 36 17

Yes that is correct. It'll look the same in operational-intelligence

**Comment by f1cognite** [comment]
@Kamillowl Should we remove the arrow or provide some children to it?

File: src/components/AssetTree/AssetTree.tsx [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+  }
+
+  mapDataAssets = (assets: AssetType[]) => {
+    const nodes: { [name: string]: TreeNodeData } = {};
+
+    assets.forEach(asset => {
+      nodes[asset.id] = this.returnPretty(asset);
+    });
+
+    const addedAsChildren: (number | string)[] = [];
```
Comment (f1cognite) [inline src/components/AssetTree/AssetTree.tsx]: Will not this always be of type `number[]`?

File: src/components/AssetTree/AssetTree.tsx [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+  onLoadData = async (treeNode: AntTreeNode) => {
+    const { loadData } = this.props;
+
+    if (treeNode.props.children) {
+      return;
+    }
+    const assetId = treeNode.props.eventKey;
+
+    if (loadData && assetId) {
+      const query = {
```
Comment (f1cognite) [inline src/components/AssetTree/AssetTree.tsx]: Here is something we can improve over time. Can we find a smarter query than alway ask for depth=2? What if the result for depth=2 is greater than 1000? Then we loose some data.
The API / SDK isn't perfect here but something to keep in mind to later.

File: src/components/AssetTree/AssetTree.tsx (hunk 1 above)
Comment (Kamillowl) [reply src/components/AssetTree/AssetTree.tsx]: @f1cognite we push asset.id to this - which is defined as string or number - so in short - no.

if so - this is a larger change we will do. From opint id is defined as both and we've constructed our types to match :)

**Pull request merged**